channeldb/migration14/migration.go

```diff
@@ -0,0 +1,90 @@
+package migration14
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/coreos/bbolt"
+)
+
+// MigrateCommitDiff migrates the CommitDiff struct on-disk for all active
+// channels to store the CommitSig with a length prefix, just as we do on the
+// wire. This ensures we'll be able to store any extra data appended to the
+// message, yet still serialize items after the message as well on disk.
+func MigrateCommitDiff(tx *bbolt.Tx) error {
+	openChanBucket := tx.Bucket(openChannelBucket)
+	if openChanBucket == nil {
+		return nil
+	}
+
+	return openChanBucket.ForEach(func(nodePub, v []byte) error {
+		// Ensure that this is a key the same size as a pubkey,
+		// and also that it leads directly to a bucket.
+		if len(nodePub) != 33 || v != nil {
+			return nil
```

Comment:
shouldn't it be an error if it doesn't? or are there other keys in the bucket?

Reply:
This was copied from our current logic where it served as more of a sanity check. Can add an error, could possibly surface issues with legacy DBs perhaps.